### ISSUE TITLE
[Enhancement] Allow setting NetworkPolicy for Metrics Port

### ISSUE THREAD

**Issue by forsberg** [strimzi/strimzi-kafka-operator#3467]
I would like to isolate the metrics port (9404) so that only Prometheus can talk to it. As far as I can see, there's no built in support in the Kafka CRD for doing so, although I can of course do it by adding a separate NetworkPolicy.

**Comment by davidjlynn** [comment]
This would also be valuable for me.

How to do this, I was hoping that [NetworkPolicyPeers](https://strimzi.io/docs/operators/latest/using.html#configuration-listener-network-policy-reference), which is used to restrict the regular listeners, could be used when we define metrics.

We define metrics here: https://strimzi.io/docs/operators/latest/using.html#type-JmxPrometheusExporterMetrics-reference
So perhaps another property `networkPolicyPeers` to align with the listeners would be sensible.

Note there is also the deprecated metrics `metrics` rather than the replacement of `metricsConfig`, we can likely disregard this when adding this feature, as the migration required to get this enhancement does not seem particularly hard.

**Comment by scholzj** [comment]
TBH, I'm not sure where it fits in the API. It might fit into the template section maybe. Something I also considered was to have a feature gate to just disable network policy creation and let user choose if they want to manage all network policies on their own (I think the metrics as they are right now work fine for 9X% people - but some people have different some special requirements and it might be hard to them all into the CRD APIs).

**Comment by davidjlynn** [comment]
I think a feature gate is a really good idea.
Otherwise Strimzi may get more requests for functionality around NetworkPolicy, (Like [Egress](https://github.com/strimzi/strimzi-kafka-operator/issues/2212) for example)

If a consumer is locking down access to Kafka, they are likely comfortable creating their own network policies, so disabling the network policy creation will solve the issue nicely.

I do still see benefit for Strimzi supporting the existing behaviour of by default adding "allow all to access kafka" network policies, it gets Kafka up and running even in a deployment with a deny-all network policy.

Then if a consumer is wanting granular network policies they can disable and write their own.

**Comment by scholzj** [comment]
> I do still see benefit for Strimzi supporting the existing behaviour of by default adding "allow all to access kafka" network policies, it gets Kafka up and running even in a deployment with a deny-all network policy.

Yeah, that is really one of the main reasons why we added it in the first place.

**Comment by scholzj** [comment]
Triaged on 31.3.2022: Users can now disable the network policy management completely and create their own network policies exactly as they need it. That includes managing the metrics port. This might not be as comfortable as doing it in the CR, but it leaves a lot more flexibility without adding additional complexity and maintenance overhead.
